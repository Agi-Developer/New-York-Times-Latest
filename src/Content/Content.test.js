import React from 'react';
import { act, cleanup, fireEvent, render } from 'react-testing-library';

import Content from './Content';

const data = [
	{
		id: 123,
		byline: 'by NewYork Times',
		media: [
			{
				'media-metadata': [
					{},
					{
						url: 'https://imgplaceholder.com'
					},
					{},
					{},
					{
						url: 'https://imgplaceholder.com'
					}
				]
			}
		],
		published_date: '2019-24-1',
		title: 'a title'
	},
	{
		id: 124,
		byline: 'by NewYork Times',
		media: [
			{
				'media-metadata': [
					{},
					{
						url: 'https://imgplaceholder.com'
					},
					{},
					{},
					{
						url: 'https://imgplaceholder.com'
					}
				]
			}
		],
		published_date: '2019-24-1',
		title: 'a title 2'
	}
];

describe('Content', () => {
	let spy;

	beforeAll(() => {
		spy = jest
			.spyOn(window, 'scrollTo')
			.mockImplementation(() => (window.scrollY = 0));
	});

	afterEach(cleanup);

	it('should render error if exists', () => {
		const { container } = render(<Content data={data} error="error occured" />);

		expect(container.querySelector('.error').textContent).toBe('error occured');
	});

	it('should display content when data is passed', () => {
		const { container } = render(<Content data={data} />);

		expect(container.firstChild).toMatchSnapshot();
	});

	describe('OnClick', () => {
		let container;

		beforeEach(() => {
			({ container } = render(<Content data={data} />));
		});

		afterEach(cleanup);

		it('should set first item selected on click', () => {
			const contentItems = container.querySelectorAll('.content--item');
			fireEvent.click(contentItems[0]);
			expect(
				container.querySelector('.detail .detail--title').textContent
			).toBe('a title');
		});

		it('should set second item selected on click', () => {
			const contentItems = container.querySelectorAll('.content--item');
			fireEvent.click(contentItems[1]);
			expect(
				container.querySelector('.detail .detail--title').textContent
			).toBe('a title 2');
		});
	});
});
