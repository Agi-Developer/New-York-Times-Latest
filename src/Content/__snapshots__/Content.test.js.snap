// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should display content when data is passed 1`] = `
<main
  class="content"
>
  <div
    class="content--item"
  >
    <div
      class="content--image"
    >
      <img
        alt="thumbnail"
        src="https://imgplaceholder.com"
      />
    </div>
    <div
      class="content--body"
    >
      <div
        class="content--body-title"
      >
        a title
      </div>
      <div
        class="content--footer"
      >
        <div
          class="content--footer-byline"
        >
          by NewYork Times
        </div>
        <div
          class="content--footer-date"
        >
          <svg
            height="15"
            viewBox="0 0 25 25"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          2019-24-1
        </div>
      </div>
    </div>
    <div
      class="content--item-right"
    >
      <svg
        height="24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </div>
  <div
    class="content--item"
  >
    <div
      class="content--image"
    >
      <img
        alt="thumbnail"
        src="https://imgplaceholder.com"
      />
    </div>
    <div
      class="content--body"
    >
      <div
        class="content--body-title"
      >
        a title 2
      </div>
      <div
        class="content--footer"
      >
        <div
          class="content--footer-byline"
        >
          by NewYork Times
        </div>
        <div
          class="content--footer-date"
        >
          <svg
            height="15"
            viewBox="0 0 25 25"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 3h-1V1h-2v2H7V1H5v2H4c-1.1 0-2 .9-2 2v16c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm0 18H4V8h16v13z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
          2019-24-1
        </div>
      </div>
    </div>
    <div
      class="content--item-right"
    >
      <svg
        height="24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
        />
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
      </svg>
    </div>
  </div>
</main>
`;
